typescript/servicecatalog/portfolio-with-ec2-product/lib/portfolio-with-ec2-product.ts

```diff
@@ -0,0 +1,122 @@
+import { Construct } from 'constructs';
+import { Stack, StackProps, Fn, CfnOutput, CfnParameter } from 'aws-cdk-lib';
+import { aws_iam as iam, aws_ec2 as ec2, aws_sns as sns } from 'aws-cdk-lib';
+import * as sc from '@aws-cdk/aws-servicecatalog-alpha';
+
+class Ec2CdkProductStack extends sc.ProductStack {
+  constructor(scope: Construct, id: string) {
+    super(scope, id);
+
+    const vpc = new ec2.Vpc(this, 'VPC', {
+      natGateways: 0,
+      subnetConfiguration: [{
+        cidrMask: 24,
+        name: 'public',
+        subnetType: ec2.SubnetType.PUBLIC,
+      }]
+    });
+
+    // Allow SSH (TCP Port 22) access from anywhere
+    const securityGroup = new ec2.SecurityGroup(this, 'SecurityGroup', {
+      vpc,
+      description: 'Allow SSH (TCP port 22) in',
+      allowAllOutbound: true,
+    });
+    securityGroup.addIngressRule(ec2.Peer.anyIpv4(), ec2.Port.tcp(22), 'Allow SSH Access')
+
+    const role = new iam.Role(this, 'ec2Role', {
+      assumedBy: new iam.ServicePrincipal('ec2.amazonaws.com'),
+    });
+
+    role.addManagedPolicy(iam.ManagedPolicy.fromAwsManagedPolicyName('AmazonSSMManagedInstanceCore'))
+
+    // Use Latest Amazon Linux Image - CPU Type ARM64
+    const ami = new ec2.AmazonLinuxImage({
+      generation: ec2.AmazonLinuxGeneration.AMAZON_LINUX_2,
+      cpuType: ec2.AmazonLinuxCpuType.ARM_64,
+    });
+
+    // EC2 Instance Type parameter
+    const ec2InstanceType = new CfnParameter(this, 'InstanceType', {
+      type: 'String',
+      description: 'The instance type of an EC2 instance.',
+    });
+
+    // Create the instance using the Security Group, AMI, and KeyPair defined in the VPC created
+    const ec2Instance = new ec2.Instance(this, 'Instance', {
+      vpc,
+      instanceType: new ec2.InstanceType(ec2InstanceType.valueAsString),
+      machineImage: ami,
+      securityGroup: securityGroup,
+      role: role,
+    });
```

Comment:
```suggestion
    const ec2Instance = new ec2.Instance(this, 'Instance', {
      vpc,
      instanceType: new ec2.InstanceType(ec2InstanceType.valueAsString),
      machineImage: ami,
      allowAllOutbound: true,
      role: role,
    });
    ec2Instance.connections.allowFromAnyIpv4(22, 'Allow SSH (TCP port 22) in');
```

Same suggestion as the python example. Use connections before reaching for security groups.